php/class-wp-cli.php

```diff
@@ -45,6 +45,145 @@ public function create_guest_authors( $args, $assoc_args ) {
+	 * @synopsis <guest-author-ids-file> [--dry-run]
+	 */
+	public function delete_guest_authors( $args, $assoc_args ) {
+		global $coauthors_plus;
+
+		$defaults = array(
+			'dry-run' => true,
+		);
+
+		$assoc_args = wp_parse_args( $assoc_args, $defaults );
+
+		$dry_run = $assoc_args['dry-run'];
+
+		// Safety first.
+		$filename_status = validate_file( $args[0] );
+
+		if ( 0 !== $filename_status ) {
+			WP_CLI::error( __( 'The provided file cannot be accessed. Please use either relative or absolute pathnames, and avoid using directory traversal such as ../ and other methods.', 'co-authors-plus' ) );
+		}
+
+		$filename = $args[0];
+
+		// Add custom error handling on fopen error.
+		// Traditionally, fopen failures result in E_WARNING messages being logged to the
+		// console. But those are really ugly and disquieting for end users to see, and usually
+		// it makes them think something is wrong with the code, rather than the simple error
+		// of "this file isn't available / isn't readable by this user / etc".
+		// So I decided to pass the message through wp-cli's display API to make it look nicer.
+		set_error_handler(
+			function( $errno, $errstr, $errfile, $errline, array $errcontext ) {
+				WP_CLI::error( sprintf(
+					/* translators: %s: Error message */
+					__( 'Error while attempting to open file: %s', 'co-authors-plus' ),
+					$errstr
+				) );
+			},
+			E_WARNING
+		);
+
+		$file = fopen( $filename, 'rb' );
+
+		// Reset normal behaviour.
+		restore_error_handler();
+
+		// Are you really, really sure.
+		if ( false === $dry_run ) {
+			WP_CLI::confirm( __( 'You have provided the --no-dry-run flag. Please confirm you wish to delete guest authors.', 'co-authors-plus' ) );
+		} else {
+			WP_CLI::log( __( 'DRY RUN', 'co-authors-plus' ) );
+		}
+
+		while ( ( $line = fgetcsv( $file, 100, ',' ) ) !== false ) {
+			$id       = filter_var( $line[0], FILTER_SANITIZE_NUMBER_INT );
+			$assignee = filter_var( $line[1], FILTER_SANITIZE_STRING );
+
+			$guest_author = $coauthors_plus->guest_authors->get_guest_author_by( 'ID', $id );
+
+			if ( true === $dry_run ) {
+
+				if ( false === $guest_author ) {
+					WP_CLI::log( sprintf(
+						/* translators: 1: Guest author ID 2: Intended assignee */
+						__( 'Attempting to delete guest author %d, but they do not exist.', 'co-authors-plus' ),
+						$id
+					) );
+				} else {
+					WP_CLI::log( sprintf(
+						/* translators: 1: Guest author ID 2: Intended assignee 3: Guest author display name 4: Guest author email 5: Guest author linked account */
+						__( 'Attempting to delete guest author %1$d, intended assignee %2$s. Display Name: %3$s Email: %4$s Linked Account: %5$s', 'co-authors-plus' ),
+						$id,
+						$assignee,
+						$guest_author->display_name,
+						$guest_author->user_email,
+						$guest_author->linked_account
+					) );
+				}
+			} else {
+				// Have to cast so the function won't literally try and reassign to user "false".
+				if ( 'false' === $assignee ) {
```

Comment:
What happens if someone has the username "false"? Or "null" or similar?

Reply:
Haha, you tricky trickster. I've added code to deal with that now :+1: